src/core/qgsvectorfilewriter.h

```diff
@@ -185,6 +185,16 @@ class CORE_EXPORT QgsVectorFileWriter : public QgsFeatureSink
       SymbolLayerSymbology //Exports one feature per symbol layer (considering symbol levels)
     };
 
+    /**
+     * Source for exported field names.
+     *
+     * \since QGIS 3.18
+     */
+    enum FieldNameSource
+    {
+      Original = 0, //!< Use original field names
+      PreferAlias, //!< Use the original field alias as the exported field name, wherever one is set. Otherwise use the original field names.
```

Comment:
```suggestion
      PreferAlias, //!< Use the field alias as the exported field name, wherever one is set. Otherwise use the original field names.
```